Package the sources necessary to produce Javadocs

jjaderberg opened the pull request:
When the fullBuild profile was removed some modules no longer produced
the sources-jar necessary in the downstream Javadocs build.
This adds `maven-source-plugin` to these modules directly so that
a sources-jar is produced by default as part of the `package` phase.

lassewesth commented:
@jjaderberg Why is this not just one single declaration at the root level?

jjaderberg commented:
@lassewesth Because we don't need a `sources-jar` from every module

lassewesth commented:
@jjaderberg Let's just build a sources jar for every module and keep the poms simple. Doing it per-module seems brittle

jjaderberg committed:
Package the sources necessary to produce Javadocs

When the fullBuild profile was removed some modules no longer produced
the sources-jar necessary in the downstream Javadocs build.
This adds maven-source-plugin to these modules directly so that
a sources-jar is produced by default as part of the package phase.

jjaderberg commented:
@lassewesth OK, I've moved it to the root pom.
